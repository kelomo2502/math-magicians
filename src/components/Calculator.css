.container {
  font-size: 1.5em;
  background: #d1d1d1;
  width: 36rem;
  height: 28rem;
  padding: 0.125rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr 1fr 1fr 1fr;
  gap: 2px 2px;
  grid-auto-flow: row;
  grid-template-areas:
    "result result result result"
    "btn-clear btn-toogle-sign btn-percentage btn-divider"
    "btn-seven btn-eight btn-nine btn-multiply"
    "btn-four btn-five btn-six btn-substract"
    "btn-one btn-two btn-three btn-plus"
    "btn-zero btn-zero btn-dot btn-equal";
}

.result {
  background: #858694;
  grid-area: result;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  color: white;
  font-size: 2em;
  padding: 0.3rem;
}

.btn-zero {
  grid-area: btn-zero;
}

.btn-dot { grid-area: btn-dot; }

.btn-equal { grid-area: btn-equal; }

.btn-clear { grid-area: btn-clear; }

.btn-toogle-sign { grid-area: btn-toogle-sign; }

.btn-percentage { grid-area: btn-percentage; }

.btn-divider { grid-area: btn-divider; }

.btn-multiply { grid-area: btn-multiply; }

.btn-substract { grid-area: btn-substract; }

.btn-plus { grid-area: btn-plus; }

.btn-one { grid-area: btn-one; }

.btn-two { grid-area: btn-two; }

.btn-three { grid-area: btn-three; }

.btn-four { grid-area: btn-four; }

.btn-five { grid-area: btn-five; }

.btn-six { grid-area: btn-six; }

.btn-seven { grid-area: btn-seven; }

.btn-eight { grid-area: btn-eight; }

.btn-nine { grid-area: btn-nine; }
