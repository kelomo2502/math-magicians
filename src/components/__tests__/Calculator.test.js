import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';
import '@testing-library/jest-dom/extend-expect';

describe('Test render Calculator component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test user event with Button in Calculator component', () => {
  it('Result should be 6', () => {
    render(<Calculator />);
    const btnElement = screen.getByText(/6/i);
    const resultElement = screen.getByTestId('result-id');
    fireEvent.click(btnElement);
    expect(resultElement).toHaveTextContent('6');
  });
  it('Result should be 53', () => {
    render(<Calculator />);
    const btn1Element = screen.getByText(/5/i);
    const btn2Element = screen.getByText(/3/i);
    const resultElement = screen.getByTestId('result-id');
    fireEvent.click(btn1Element);
    fireEvent.click(btn2Element);
    expect(resultElement).toHaveTextContent('53');
  });
});
