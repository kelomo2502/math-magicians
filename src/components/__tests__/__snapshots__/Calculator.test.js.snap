// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render Calculator component renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "margin": "0 auto",
      "width": "56rem",
    }
  }
>
  <h1
    style={
      Object {
        "color": "#00c2cb",
        "fontFamily": "Poppins",
        "margin": "0",
      }
    }
  >
    Lets do some Math
  </h1>
  <div
    className="container"
  >
    <div
      className="result"
      data-testid="result-id"
    />
    <button
      className="sc-bczRLJ kLLeMr btn-zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="sc-bczRLJ foUeaX btn-equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-clear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-toogle-sign"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-percentage"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="sc-bczRLJ foUeaX btn-divider"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
    <button
      className="sc-bczRLJ foUeaX btn-multiply"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="sc-bczRLJ foUeaX btn-substract"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="sc-bczRLJ foUeaX btn-plus"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-one"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-two"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-three"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-four"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-five"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-six"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-seven"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-eight"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-nine"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="sc-bczRLJ kLLeMr btn-dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
  </div>
</div>
`;
